File: 5. Fifth_Semester/PYTHON/Minor Assignment 2/Q22.py
n = int(input("Enter a Positive Integer :"))

def sumDigits(n):
    total = 0
    while(n > 0):
        total += n % 10
        n //= 10
    return total

while(n >= 10):
    n = sumDigits(n)
    
print("The Single digit sum is :", n)

File: 5. Fifth_Semester/C LANGUAGE (PPWC)/Minor Assignment 04/Q22.c
#include <stdio.h>

void sum(int a, int b, int *cp) {
    *cp = a + b;
}

int main() {
    int x = 7, y = 2, z;

    printf("x y z\n");
    sum(x, y, &z);
    printf("%4d%4d%4d\n", x, y, z);
    sum(y, x, &z);
    printf("%4d%4d%4d\n", x, y, z);
    sum(z, y, &x);
    printf("%4d%4d%4d\n", x, y, z);
    sum(z, z, &x);
    printf("%4d%4d%4d\n", x, y, z);
    sum(y, y, &y);
    printf("%4d%4d%4d\n", x, y, z);

    return 0;
}
